### ISSUE TITLE
Possible issue with :source-uri option?

### ISSUE THREAD

**Issue by ptaoussanis** [weavejester/codox#118]
Hi James,

Am running Codox 0.9.4 with Clojure 1.8, project.clj config is:
```clojure
  :codox
  {:language :clojure
   :source-paths ["target/classes"]
   :source-uri
   {#"target/classes" "https://github.com/ptaoussanis/sente/blob/master/src/{classpath}x#L{line}"
    #".*" "https://github.com/ptaoussanis/sente/blob/master/{filepath}#L{line}"}}
```

This matches up closely to the current example in the README:
```clojure
:codox
{:source-uri
 {#"target/classes" "https://github.com/foo/bar/blob/master/src/{classpath}x#L{line}"
  #".*"             "https://github.com/foo/bar/blob/master/{filepath}#L{line}"}
```

A user just reported that the source links are broken, generating links like
`https://github.com/ptaoussanis/sente/blob/master/target/classes/taoensso/sente.clj#L102`
instead of 
`https://github.com/ptaoussanis/sente/blob/master/src/taoensso/sente.cljx#L102` (which I'd expect with the given config).

On the road atm so only spent a minute looking at this, but removing the `#".*"` config entry seems to correct the links. Given that the source-uri Regexes are provided through map keys (unordered), I'm wondering if the issue might be a linear search somewhere like this: https://github.com/weavejester/codox/blob/master/codox/src/codox/writer/html.clj#L119

(Just skimmed the code quick; no idea if this particular code path is relevant).

Have I got the config wrong? Or, assuming this could be an ordering issue, any chance we could instead specify :source-uri variations as a vector?

Thanks a lot for your assistance + work on Codox, cheers!

**Comment by weavejester** [comment]
Yes, this looks like an ordering bug that slipped my mind when I wrote the README. I guess the solution would be to allow an ordered data structure like a vector to be used in place of a map.
